tests/unit/dcc_comms/test_mqtt_dcc_comms.py

```diff
@@ -0,0 +1,257 @@
+        self.client = None
+
+    @mock.patch.object(Mqtt, 'connect_soc')
+    def test_init_no_msgattr_no_client_id(self, mock_connect):
+        """
+        Test MqttDccComms initialisation without messaging attributes and no client ID
+        :param mock_connect: Mocked MQTT connect method
+        :return:
+        """
+        MqttDccComms(edge_system_name=self.edge_system.name,
+                     url=self.url, port=self.port,
+                     clean_session=self.clean_session)
+
+        # Read uuid.ini
+        self.config.read(self.uuid_file)
+
+        edge_system_name = self.config.get('GATEWAY', 'name')
+        local_uuid = self.config.get('GATEWAY', 'local-uuid')
+
+        self.assertEquals(edge_system_name, self.edge_system.name)
+        self.assertIsNotNone(local_uuid)
+        mock_connect.assert_called()
+
+    @mock.patch.object(Mqtt, 'connect_soc')
+    def test_init_no_msgattr_with_client_id(self, mock_connect):
+        """
+        Test MqttDccComms initialisation without messaging attributes and non-empty client ID
+        :param mock_connect: Mocked MQTT connect method
+        :return:
+        """
+        MqttDccComms(edge_system_name=self.edge_system.name,
+                     url=self.url, port=self.port,
+                     client_id="unit-test", clean_session=self.clean_session)
+
+        # Read uuid.ini
+        self.config.read(self.uuid_file)
+
+        edge_system_name = self.config.get('GATEWAY', 'name')
+        local_uuid = self.config.get('GATEWAY', 'local-uuid')
+
+        self.assertEquals(edge_system_name, self.edge_system.name)
+        self.assertEquals(local_uuid, "unit-test")
+        mock_connect.assert_called()
+
+    @mock.patch.object(Mqtt, 'connect_soc')
+    def test_init_with_valid_msgattr_with_client_id(self, mock_connect):
+        """
+        Test MqttDccComms initialisation with valid messaging attributes and non-empty client id
+        :param mock_connect: Mocked MQTT connect method
+        :return:
+        """
+
+        # Test with valid messaging attributes
+        MqttDccComms(edge_system_name=self.edge_system.name,
+                     url=self.url, port=self.port, client_id="unit-test", clean_session=self.clean_session,
+                     mqtt_msg_attr=self.mqtt_msg_attr)
+
+        mock_connect.assert_called()
+
+    @mock.patch.object(Mqtt, 'connect_soc')
+    def test_init_with_valid_msgattr_empty_client_id(self, mock_connect):
+        """
+        Test MqttDccComms initialisation with valid messaging attributes with empty client id
+        :param mock_connect: Mocked MQTT connect method
+        :return:
+        """
+
+        # Test with valid messaging attributes
+        MqttDccComms(edge_system_name=self.edge_system.name,
+                     url=self.url, port=self.port, client_id="", clean_session=self.clean_session,
+                     mqtt_msg_attr=self.mqtt_msg_attr)
+
+        # Read uuid.ini
+        self.config.read(self.uuid_file)
```

Comment:
Wrong test case.  Autogeneration and writing to file happens only when mqtt_msg_attr is None.  This test case succeeds because, uuid.ini file is not deleted during tearDown()

Please add the test cases for these as mentioned in previous review:

- MqttDccComms called with edge_system_name, msg_attr=valid msg_attr and no client id. Here test for client_id is ""

- MqttDccComms called with edge_system_name, msg_attr=valid msg_attr and client id. Here test for client_id is user-provided